.github/workflows/update-locked-env.yml

```diff
@@ -0,0 +1,100 @@
+name: Update locked envs
+on:
+  push:
+    paths:
+    - envs/environment.yaml
+  schedule:
+  - cron: "0 8 1,16 * *" # Bi-weekly
+  workflow_dispatch:
+
+env:
+  BASE_ENV: envs/environment.yaml
+
+jobs:
+  update-locked-environment:
+    if: ${{ github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch' }}
```

Comment:
A very minor style comment: would be great to replace single quotes with double ones for consistency. [My bad originally, sorry!]

Reply:
Update: I'm reverting the last commit about these changes.
A ton of messages like: Unexpected symbol: '"workflow_dispatch"' appear. It is not worthy harmonizing the apexes as in some (many) places they are actually needed as is. It may be better to harmonize with single apex maybe but it becomes a maze it is not really worth exploring.

Reply:
Thanks for checking!

It appears, github actions imply using single quotes according to the [official manual](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions) and according to this [github-runners](https://github.com/actions/runner/issues/866) issue there is no intention to change it.

Not sure why double quotes are working in other places, but bbsolutely agree that it's not worth to put too much efforts there!